### PR TITLE
Linus: adjust to new schema

### LINKED ISSUE #97: db/entities: remove max-length restriction for String fields
Default seems to be varchar(255), which especially for note fields could be too small.

Someone ^TM should find out how to raise this limit globally, probably even unlimited.

I don't want to specify a max. length for every attribute separately, if possible.

### LINKED ISSUE #168: linus: adjust to newer database schema
We sadly got, despite asking explicitly, an old DB dump when specifying our Linus connection in #40 

A current dump has the following differences (only looked at the four tables we clone with our `Linus*` entity types:

* `reservation` has an additional column 
  <pre>`message` longtext DEFAULT NULL</pre>
* `experiment` has two additional columns
  <pre>`tags` longtext DEFAULT NULL,
  `presented` tinyint(1) NOT NULL,</pre>
* `reservation_experiment` **misses one column**: `user_id` is not present anymore...
* Table `user` is not present anymore. Instead, there is a table `accounts_user`:
  <pre>  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `password` varchar(128) NOT NULL,
  `last_login` datetime(6) DEFAULT NULL,
  `is_superuser` tinyint(1) NOT NULL,
  `username` varchar(150) NOT NULL,
  `first_name` varchar(150) NOT NULL,
  `last_name` varchar(150) NOT NULL,
  `is_staff` tinyint(1) NOT NULL,
  `is_active` tinyint(1) NOT NULL,
  `date_joined` datetime(6) NOT NULL,
  `email` varchar(254) NOT NULL,
  `color_id` bigint(20) DEFAULT NULL,
  `preset_id` bigint(20) DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `username` (`username`),
  KEY `accounts_user_color_id_f5527965_fk_labels_color_id` (`color_id`),
  KEY `accounts_user_preset_id_8b3084cf_fk_accounts_permissionpreset_id` (`preset_id`),
  CONSTRAINT `accounts_user_color_id_f5527965_fk_labels_color_id` FOREIGN KEY (`color_id`) REFERENCES `labels_color` (`id`),
  CONSTRAINT `accounts_user_preset_id_8b3084cf_fk_accounts_permissionpreset_id` FOREIGN KEY (`preset_id`) REFERENCES `accounts_permissionpreset` (`id`)</pre>
  
  (schema of `user` before was
  <pre>`id` int(11) NOT NULL AUTO_INCREMENT,
  `name` varchar(180) NOT NULL,
  `roles` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin NOT NULL,
  `password` varchar(255) NOT NULL,
  `email` varchar(255) NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `UNIQ_8D93D6495E237E06` (`name`)</pre>

### PR DESCRIPTION
**Pull request by JohaJung**
This changes our backend entity schema to fit to the *actual* production Linus instance.

I moved a working sample-data script to our internal hessenbox, you can fill your linus-db container for testing using a command like this one (adjust the file path if necessary), run from project root folder:

```bash
cat ../hessenbox/testdata-internal/testdata-internal-corrected.sql | docker compose exec -T linus-db mysql db_name --user=db_user --password=db_password
```

closes #168 

While I was at the backend entities, this now also fixes #97 :) (Yes, I'm to blame for mixing topics in this PR ^^)

---

### Implementation

- [ ] Does this code change accomplish what it is supposed to do?
- [ ] Can this solution be simplified without unreasonable effort
- [ ] Is the code modular enough?
- [ ] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [ ] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [ ] Interface Segregation: If interface implemented, is it fully implemented?
- [ ] Dependency Injection: Are central object instantiation processes managed by a single context?
- [ ] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [ ] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [ ] Is error handling done the correct way?
- [ ] Should any logging or debugging information be added or removed?
- [ ] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [ ] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [ ] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [ ] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by smaller methods/different names?
- [ ] Is the code located in the right file/folder?
- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message better?
- [ ] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

**Comment by JohaJung** [comment]
Despite that our new example dataset contains up-to-date experiment bookings, I did not manage the linus sync button to recognize the booked experiments...
@MarcoLeipnitz do you have time to take a look at this? Können wir gerne gemeinsam machen.

Following are the newest bookings that exist:

```
MariaDB [db_name]> SELECT * FROM reservation_experiment ORDER BY id DESC LIMIT 10;
+------+--------+----------------------------+---------------+----------------+
| id   | status | pinned_on                  | experiment_id | reservation_id |
+------+--------+----------------------------+---------------+----------------+
| 2102 |      1 | 2026-03-09 10:20:29.589402 |           885 |            621 |
| 2101 |      1 | 2026-03-09 10:12:22.412419 |           555 |            620 |
| 2100 |      1 | 2026-03-09 10:12:22.412419 |           484 |            620 |
| 2099 |      1 | 2026-02-11 08:25:19.147972 |           716 |            619 |
| 2098 |      1 | 2026-02-11 08:25:19.147972 |           723 |            619 |
| 2097 |      1 | 2026-02-08 14:17:19.978207 |            85 |            618 |
| 2096 |      1 | 2026-02-08 14:17:19.978207 |            79 |            618 |
| 2095 |      1 | 2026-02-08 14:17:19.978207 |           654 |            618 |
| 2094 |      1 | 2026-02-08 14:09:32.386214 |           840 |            617 |
| 2093 |      1 | 2026-02-03 07:47:14.976465 |           669 |            616 |
+------+--------+----------------------------+---------------+----------------+
10 rows in set (0.000 sec)

MariaDB [db_name]> SELECT * FROM reservation ORDER BY id DESC LIMIT 6;
+-----+----------------------------+--------+----------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------+----------------------------------+---------+
| id  | ordertime                  | status | lecture_date               | comment                                                                                                                                                            | message                                                                                                   | name                             | user_id |
+-----+----------------------------+--------+----------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------+----------------------------------+---------+
| 621 | 2026-03-09 10:20:29.589402 |      1 | 2026-03-12 10:40:00.000000 |                                                                                                                                                                    |                                                                                                           | Florian Test                     |       3 |
| 620 | 2026-03-09 10:12:22.412419 |      1 | 2026-03-10 08:50:00.000000 | Bitte das wurfmikro vorbereiten                                                                                                                                    | Komme zur Abnahme 5min vorher                                                                             | Test Florian                     |       3 |
| 619 | 2026-02-11 08:25:19.147972 |      1 | 2026-02-13 09:10:00.000000 | Bei O.A.4.1 nur Strahlengänge von mehreren anfänglich parallelen Strahlen weissen Lichts durch Linsen.                                                             | Bei O.A.4.1 nur Strahlengänge von mehreren anfänglich parallelen Strahlen weissen Lichts durch Linsen.    | Physik für Biologie, 13.02.2026  |      19 |
| 618 | 2026-02-08 14:17:19.978207 |      1 | 2026-02-12 07:00:00.000000 | Lecherleitung dient als Werbung für das nächste Semester: Nachweis elektromagnetischer Wellen. 
Mit Kondensatorring und Glühbirnchen und mit Leuchtstoffröhre     |                                                                                                           | Physik-I Pietralla               |      37 |
| 617 | 2026-02-08 14:09:32.386214 |      1 | 2026-02-10 07:00:00.000000 | Es reicht, den Murmelturm als reverse Video anzusehen und danach auch einmal in richtiger Zeitrichtung.                                                            |                                                                                                           | Physik-I Pietralla               |      37 |
| 616 | 2026-02-03 07:47:14.976465 |      1 | 2026-02-06 09:00:00.000000 |                                                                                                                                                                    |                                                                                                           | Physics for Biologists           |      42 |
+-----+----------------------------+--------+----------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------+----------------------------------+---------+
6 rows in set (0.000 sec)
```

**Comment by JohaJung** [comment]
The last commit fixes an important hidden bug: only whole-day appointments are shown for matching, on current mayn.

<details>
<summary>Fix:</summary>

```diff
diff --git a/VLA-Frontend/src/components/linusAppointmentMatcher/AppointmentMatchEntry.tsx b/VLA-Frontend/src/components/linusAppointmentMatcher/AppointmentMatchEntry.tsx
index 2feadd4..bb31899 100644
--- a/VLA-Frontend/src/components/linusAppointmentMatcher/AppointmentMatchEntry.tsx
+++ b/VLA-Frontend/src/components/linusAppointmentMatcher/AppointmentMatchEntry.tsx
@@ -51,9 +51,9 @@ export default function AppointmentMatchEntry({
 
   if(dayAppointments !== undefined) {
     const nonWholeDayAppointments = dayAppointments.filter(
-      appointment => isUntimedForView(appointment, calendarStartMinSinceMidnight, calendarEndMinSinceMidnight)
+      appointment => !isUntimedForView(appointment, calendarStartMinSinceMidnight, calendarEndMinSinceMidnight)
     );
-    
+
     const linusAppointmentInTimeFrameAppointments = nonWholeDayAppointments.filter(appointment => 
       appointment.startTime.getTime() <= matching.linusAppointmentTime.getTime() &&
       matching.linusAppointmentTime.getTime() <= appointment.endTime.getTime()
```

</details>

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/entities/linusdb/LinusUser.java [hunk 1]
```diff
@@ -22,26 +24,76 @@ public class LinusUser {
...
+    /**
+     * Timestamp of the last login.
+     */
+    @Column(name = "last_login")
+    @Nullable
+    private LocalDateTime lastLogin;
+
+    /**
+     * Timestamp of account creation.
+     */
```
Comment (MarcoLeipnitz) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/entities/linusdb/LinusUser.java:65]: Missing annotation

**Review by MarcoLeipnitz — APPROVED** [review]
Please add the column annotation for more clarity.

**Comment by MarcoLeipnitz** [comment]
And consider removing the previous linus instance from our project, this might generate confusion for further developers and maintainers.

**Review by MarcoLeipnitz — CHANGES_REQUESTED** [review]
I have found a bug that reimports deleted experiments

Match a LinusAppointment that has an experiment with an Appointment
Delete the experimentBooking in the overview
Navigate in time
The deleted experiment is imported again

**Comment by MarcoLeipnitz** [comment]
There is an error on line 190 in Linussyncservices that states that the query result is not unique

This is no longer reproducable but it has occured somehow

**Comment by MarcoLeipnitz** [comment]
I think the non-unique result was a result of quickly navigation and starting two threads on the syncing endpoint which resulted in two copies of the experiment

We should consider adding a synchronized there

**Comment by MarcoLeipnitz** [comment]
And the notes that are stored in the experiment do not respect \n chars, you have to transform that into html `<br>` if you want newlines

**Comment by MarcoLeipnitz** [comment]
The duplication bug of experiment has probably also existed before this migration and went unnoticed, if we assume linus reservations to be static we can just check if the linusAppointment has a valid appointmentMatching and if so we can just ignore the syncing in experimentBooking.

Context: Linussyncservice, syncexperimentBooking

**Comment by JohaJung** [comment]
> The duplication bug of experiment has probably also existed before this migration and went unnoticed, if we assume linus reservations to be static we can just check if the linusAppointment has a valid appointmentMatching and if so we can just ignore the syncing in experimentBooking.
> 
> Context: Linussyncservice, syncexperimentBooking

This is a great idea imo. Otherwise I would have introduced a `deleted` bit into the experimentBooking entity

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/entities/linusdb/LinusAppointment.java
Comment (JohaJung) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/entities/linusdb/LinusAppointment.java:64]: Note to myself: Try to fix timezone issues by overriding getter method for this attribute and handling that there

**Comment by MarcoLeipnitz** [comment]
I have fixed the bug with the experiments being reimported.
I have not found a better solution than marking appointmentMatchings as already synced without rewriting the logic behind LinusSyncService

**Comment by MarcoLeipnitz** [comment]
About the time shift issue
From what I've quickly skimmed I can tell you that the repository methods do not respect the custom setter you built.
You should probably just adjust the few usages in the code and add something in the docstring.

**Comment by JohaJung** [comment]
> About the time shift issue From what I've quickly skimmed I can tell you that the repository methods do not respect the custom setter you built. You should probably just adjust the few usages in the code and add something in the docstring.

After an hour of debugging, I found out that they do. My timezone-fixing logic was just wrong.
